### PR TITLE
🔀New CI/CD (TC -> GHA)

### PR DESCRIPTION
**Pull request by kdev**
**What kind of update does this PR provide?** *Please check*
- [ ] new translations / updated translations / translation fixes
- [ ] a new feature
- [ ] a new module
- [ ] a bugfix for an existing feature / module
- [ ] improvement of style or user experience
- [x] other: CI 

**What is included in your update?**
* reworked workflows
* usage of GH Actions over TeamCity

**ToDo**
- [ ] dependency caching

### CODE REVIEW

**Comment by kdev** [comment]
To test this functionality it is maybe required to enable Actions for this repo? (cc @jxn-30)

**Comment by jxn-30** [comment]
Actions are already enabled but I will check if there are any restricting settings

**Comment by jxn-30** [comment]
I've updated the config, an option has been enabled that restricted actions.
Please try again.

**Comment by kdev** [comment]
I actually dont know why the workflow is not being triggered

**Review by jxn-30 — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+            - name: '[⬆] update browserslist'
+              run: npx -y browserslist@latest --update-db
+
+            - name: '[🍱] yarn install'
+              run: yarn install --immutable
+
+            - name: '[🌳] set env variables (e.g. beta/stable for webpack & target dir)'
+              run: |
+                  echo "MODE=$(if ${{ github.ref == 'refs/heads/master' }}; then echo "production"; else echo "development"; fi)" >> $GITHUB_ENV
```
Comment (Cr4zyc4k3) [inline .github/workflows/build.yml:39]: Every feature branch is build as dev branch?

File: .github/workflows/development.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+              run: ssh-keyscan -H ${{ secrets.SSH_HOST }} >> ~/.ssh/known_hosts
+
+            - name: Deployment
+              if: ${{ success() }}
+              run: |
+                  sudo apt-get install -y rsync
+
+                  rsync -run --exclude dist/static/* dist/ "/${{ needs.build.outputs.type }}" 
+                  exit 0
+              # cp -rf "$DIR_STATIC"/* "$DEPLOY_DIR"
```
Comment (Cr4zyc4k3) [inline .github/workflows/development.yml:36]: Remove if not needed

**Review by Cr4zyc4k3 — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (kdev) [reply .github/workflows/build.yml:39]: afaic yes

Comment (jxn-30) [reply .github/workflows/build.yml:39]: that is correct. Only builds on master should be stable builds.

**Pull request merged**